pyproject.toml

```diff
@@ -58,7 +58,7 @@ test = [
   "pytest",
   "pytest-mpl",
   "pytest-cov",
-  "xgboost",
+  "xgboost>=1.4",
```

Comment:
I suggest removing this line:
- I don't think this pin will have any effect, as the test suite on CI currently lets pip resolve the latest versions of all dependencies.
- Moreover, it may give other contributors the false impression that this config option is used to specify constraints on optional dependencies.

Reply:
Reverted the change. I thought it would be a good idea as anyone running the tests will require this constraint to be satisfied for the tests to pass. However, they will still be able to identify the issue from the error message should it arise.